### DISCUSSION TITLE
Why doesn't fsspec see files via HTTPS on this Open Storage Network bucket?

### DISCUSSION THREAD

**Discussion by rsignell-usgs** [fsspec/filesystem_spec#1220] [Q&A] [answered]
``` python
import fsspec
fs = fsspec.filesystem('https')
fs.ls('https://ncsa.osn.xsede.org/Pangeo')
```
returns `[]`

But if I point my web browser at that URL, I see:

![2023-03-22_08-18-22](https://user-images.githubusercontent.com/1872600/226902864-a3eb85bb-12af-4276-b033-a5910d2f508c.png)"

I guess that's because that's not a file, but a service, right?

**Comment by rsignell-usgs** [comment, marked as answer]
Doh!  This is really an S3-compatible [Open Storage Network](https://www.openstoragenetwork.org/) Bucket, so **this** is the way to explore files with fsspec:
![2023-03-22_08-24-20](https://user-images.githubusercontent.com/1872600/226904413-2e97c12a-f625-4099-93c2-7a99a73ab499.png)

**Comment by martindurant** [comment]
To answer the original question, in case people are interested: the http implementation does `ls` by loading the given URL, and looking for HTTP links with names that are like "{original_url}/...". They keys in that XML do not match that pattern, and are not HTTP links anyway.
